warehouse/clickhouse/clickhouse.go

```diff
@@ -355,7 +355,7 @@ func typeCastDataFromType(data string, datatype string) interface{} {
 	case "datetime":
 		t, err := time.Parse(time.RFC3339, data)
 		if err != nil {
-			return ""
+			return nil
```

Comment:
typeCastDataFromType ==> should be typecastDataFromType
datatype ==> should be dataType